src/client/components/vision2/camera/distance.ts

```diff
@@ -0,0 +1,40 @@
+import { computed } from 'mobx'
+import * as THREE from 'three'
+
+import { range } from '../../../../shared/base/range'
+import { Vector4 } from '../../../math/vector4'
+import { group } from '../../three/builders'
+import { Canvas } from '../../three/three'
+import { CameraParams } from '../camera/model'
+
+import { LineProjection } from './line_projection'
+
+export class DistanceViewModel {
+  constructor(
+    private readonly params: CameraParams,
+    private readonly lineProjection: LineProjection,
+    private readonly majorStepDistance: number,
+    private readonly minorLines: number,
+    private readonly maxDistance: number,
+  ) {
+  }
+
+  static of(canvas: Canvas, params: CameraParams): DistanceViewModel {
+    return new DistanceViewModel(params, LineProjection.of(canvas, params.lens), 1.0, 3, 5)
+  }
+
+  readonly distance = group(() => ({
+    children: range((this.minorLines + 1) * this.maxDistance / this.majorStepDistance).map(i => (
```

Comment:
Note that this will currently add 20 draw calls per frame until we refactor `LineProjection` 😄

Reply:
It's off by default so that's fine!